### PR TITLE
Register NCAR notebook gallery

### PR DESCRIPTION
**Pull request by bonnland**
I followed the contributing guide as best as I could.

### CODE REVIEW

**Comment by rabernat** [comment]
Thanks for your contribution.

Your PR failed because you pointed your submodule at a branch that doesn't exist (`binderbot-build`).

**Comment by bonnland** [comment]
OK, I assume you mean a branch in the notebook gallery.  I created it in my pangeo gallery fork because the instructions were not clear on that point.  

I assume at this point that I have to create the branch and then trigger the build again.

File: .gitmodules [hunk 1]
```diff
@@ -39,3 +39,7 @@
 	path = repos/xgcm/xgcm-examples
 	url = https://github.com/xgcm/xgcm-examples.git
 	branch = binderbot-built
+[submodule "repos/pangeo-gallery/ncar-gallery"]
+	path = repos/pangeo-gallery/ncar-gallery
+	url = https://github.com/NCAR/notebook-gallery.git
+	branch = binderbot-built
```
Comment (rabernat) [inline .gitmodules:45]: Yes, I'm referring to this part. If you try to visit it, you can see it doesn't exist: https://github.com/NCAR/notebook-gallery/tree/binderbot-built

**Comment by rabernat** [comment]
Are you actually using binderbot for building your notebooks? Or are you trying to point at a static gallery of notebooks?

**Comment by bonnland** [comment]
Hi Ryan, I'm trying to register the NCAR gallery to get our binderbot build to work, please see the end of this discussion:

https://github.com/NCAR/notebook-gallery/pull/66

This is all new to me and I realize I am flailing about a little trying to get this to all play nicely together...

**Comment by rabernat** [comment]
Your binderbot-built branch is identical to main. Why not just point at main instead?

**Comment by bonnland** [comment]
So I don't have the full context of what's needed here.   I was assuming that we are using binderbot as part of our continuous integration, which tests the running of our notebooks against the latest version of upstream dependencies.   Now that you mention it, that probably means we do want to point at main instead.   

Does this mean changing the submodule command, the binderbot-gallery.yaml file, or both?

**Comment by rabernat** [comment]
> So I don't have the full context of what's needed here.

It's actually pretty simple. The submodule linked in this repo should contain a series of **executed** notebooks, plus a `binder-gallery.yaml` file.

This is what binderbot will create for you, if you let it.

**Comment by rabernat** [comment]
If you are using binderbot, which it sounds like you are, you should indeed be pointing at the `binderbot-built`. The reason it did not exist is because you had not actually run binderbot on your master branch yet.

**Comment by bonnland** [comment]
Merge to "main" in notebook gallery had a timeout error for binderbot.   But we run nightly builds, so maybe the next one of those will succeed.

From your last comment, I am around 80% confident that this PR is a red herring and should be closed.

**Comment by bonnland** [comment]
OK, I have merged https://github.com/NCAR/notebook-gallery/pull/67 into main.  

Did the binderbot build work as expected?   It looks like there was a timeout error after 5 minutes:

https://github.com/NCAR/notebook-gallery/runs/1824799546?check_suite_focus=true

**Comment by bonnland** [comment]
I am going to take a chance and re-run the binderbot build, and hope that the original output is not lost, just in case the timeout error was a fluke.

**Comment by bonnland** [comment]
OK, the workflow got further along than last time.   It looks like a thumbnail image is expected and is not in the right location at the moment.

**Comment by rabernat** [comment]
This won't work until we see a `binderbot-built` branch in this list: https://github.com/NCAR/notebook-gallery/branches

binderbot will automatically create that for you if it ever finishes correctly! 🙃  Thanks a lot for your patience here.

**Comment by bonnland** [comment]
OK, I remember seeing an error in one of the builds about a thumbnail.png file missing;  I have lost track of which build log that was, but I've now placed a thumbnail image in https://github.com/NCAR/notebook-gallery.git

How do I re-trigger the build?   Am I right in thinking we have to do this step next?

**Comment by rabernat** [comment]
> How do I re-trigger the build?

You go to this page - https://github.com/NCAR/notebook-gallery/actions/runs/534868300 - an click "re-run workflow".

**Comment by bonnland** [comment]
OK, as you have mentioned before, the binderbot-built branch does not exist, and this causes the build to fail.   What is the next step, given this?   Here is the build output log:

```
Run git submodule update --init --recursive --remote
Submodule 'repos/NCAR/cesm-lens-aws' (https://github.com/NCAR/cesm-lens-aws.git) registered for path 'repos/NCAR/cesm-lens-aws'
Submodule 'repos/TomAugspurger/pangeo-dask-gateway' (https://github.com/TomAugspurger/pangeo-dask-gateway.git) registered for path 'repos/TomAugspurger/pangeo-dask-gateway'
Submodule 'repos/earthcube2020/ec20_abernathey_etal' (https://github.com/earthcube2020/ec20_abernathey_etal.git) registered for path 'repos/earthcube2020/ec20_abernathey_etal'
Submodule 'repos/pangeo-data/landsat-8-tutorial-gallery' (https://github.com/pangeo-data/landsat-8-tutorial-gallery.git) registered for path 'repos/pangeo-data/landsat-8-tutorial-gallery'
Submodule 'repos/pangeo-data/pangeo-tutorial-gallery' (https://github.com/pangeo-data/pangeo-tutorial-gallery.git) registered for path 'repos/pangeo-data/pangeo-tutorial-gallery'
Submodule 'repos/pangeo-gallery/cmip6' (https://github.com/pangeo-gallery/cmip6.git) registered for path 'repos/pangeo-gallery/cmip6'
Submodule 'repos/pangeo-gallery/example-gallery' (https://github.com/pangeo-gallery/example-gallery.git) registered for path 'repos/pangeo-gallery/example-gallery'
Submodule 'repos/pangeo-gallery/ncar-gallery' (https://github.com/NCAR/notebook-gallery.git) registered for path 'repos/pangeo-gallery/ncar-gallery'
Submodule 'repos/pangeo-gallery/physical-oceanography' (https://github.com/pangeo-gallery/physical-oceanography.git) registered for path 'repos/pangeo-gallery/physical-oceanography'
Submodule 'repos/pangeo-gallery/esip-gallery' (https://github.com/rsignell-usgs/esip-gallery.git) registered for path 'repos/rsignell-usgs/esip-gallery'
Submodule 'repos/xgcm/xgcm-examples' (https://github.com/xgcm/xgcm-examples.git) registered for path 'repos/xgcm/xgcm-examples'
Cloning into '/home/runner/work/pangeo-gallery/pangeo-gallery/repos/NCAR/cesm-lens-aws'...
Cloning into '/home/runner/work/pangeo-gallery/pangeo-gallery/repos/TomAugspurger/pangeo-dask-gateway'...
Cloning into '/home/runner/work/pangeo-gallery/pangeo-gallery/repos/earthcube2020/ec20_abernathey_etal'...
Cloning into '/home/runner/work/pangeo-gallery/pangeo-gallery/repos/pangeo-data/landsat-8-tutorial-gallery'...
Cloning into '/home/runner/work/pangeo-gallery/pangeo-gallery/repos/pangeo-data/pangeo-tutorial-gallery'...
Cloning into '/home/runner/work/pangeo-gallery/pangeo-gallery/repos/pangeo-gallery/cmip6'...
Cloning into '/home/runner/work/pangeo-gallery/pangeo-gallery/repos/pangeo-gallery/example-gallery'...
Cloning into '/home/runner/work/pangeo-gallery/pangeo-gallery/repos/pangeo-gallery/ncar-gallery'...
Cloning into '/home/runner/work/pangeo-gallery/pangeo-gallery/repos/pangeo-gallery/physical-oceanography'...
Cloning into '/home/runner/work/pangeo-gallery/pangeo-gallery/repos/rsignell-usgs/esip-gallery'...
Cloning into '/home/runner/work/pangeo-gallery/pangeo-gallery/repos/xgcm/xgcm-examples'...
Submodule path 'repos/NCAR/cesm-lens-aws': checked out 'ed2ff12b2c0b5244127ef04d6765dd26fa980b78'
Submodule path 'repos/TomAugspurger/pangeo-dask-gateway': checked out 'd5f6daafd9d41e3acebce4615b60bf48f021b413'
Submodule path 'repos/earthcube2020/ec20_abernathey_etal': checked out '4ca8b56dd04f101798e2831725e7f8c60e642861'
Submodule path 'repos/pangeo-data/landsat-8-tutorial-gallery': checked out 'd807776f5073b5edf3783fa55429b46146c569b9'
Submodule path 'repos/pangeo-data/pangeo-tutorial-gallery': checked out '391da4df6d24b162d20ad47d9ad16d02ccf1bb5e'
Submodule path 'repos/pangeo-gallery/cmip6': checked out 'e2d279b697cf6c8ae3e5fc720b0da86316762fb1'
Submodule path 'repos/pangeo-gallery/example-gallery': checked out 'c3457501438519442d4110e527ff50af5ced20fb'
fatal: Needed a single revision
Unable to find current origin/binderbot-built revision in submodule path 'repos/pangeo-gallery/ncar-gallery'
Error: Process completed with exit code 1.
```

**Comment by rabernat** [comment]
> OK, as you have mentioned before, the binderbot-built branch does not exist, and this causes the build to fail.

The binderbot-built branch does exist! https://github.com/NCAR/notebook-gallery/branches

Consequently, I was able to rerun the workflow here, and everything worked right! 🎉

**Comment by rabernat** [comment]
So things kind of worked.

http://gallery.pangeo.io/repos/pangeo-gallery/ncar-gallery/

Some of the links are broken, and I think I understand why.

File: .gitmodules [hunk 2]
```diff
@@ -39,3 +39,7 @@
 	path = repos/xgcm/xgcm-examples
 	url = https://github.com/xgcm/xgcm-examples.git
 	branch = binderbot-built
+[submodule "repos/pangeo-gallery/ncar-gallery"]
+	path = repos/pangeo-gallery/ncar-gallery
```
Comment (rabernat) [inline .gitmodules:43]: This is wrong. It shoudl have been `repos/NCAR/ncar-gallery`. It needs to match the github repo https://github.com/NCAR/notebook-gallery. I'll fix it.

**Pull request merged**